### PR TITLE
Don't throw when re-registering a plugin

### PR DESCRIPTION
**Pull request by misteroneill**
Currently, if a plugin is re-registered, it will throw an error. On the face, this makes sense, but it comes with two issues:

1. It will totally break in an unrecoverable way in a case where it might not actually be a break.
2. Nothing else has this behavior - components and techs do not throw.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/plugin.js [hunk 1]
```diff
@@ -305,8 +306,10 @@ class Plugin {
       throw new Error(`Illegal plugin name, "${name}", must be a string, was ${typeof name}.`);
     }
 
-    if (pluginExists(name) || Player.prototype.hasOwnProperty(name)) {
-      throw new Error(`Illegal plugin name, "${name}", already exists.`);
+    if (pluginExists(name)) {
+      log.warn(`A plugin named "${name}" already exists. You may want to avoid re-registering plugins!`);
+    } else if (Player.prototype.hasOwnProperty(name)) {
```
Comment (misteroneill) [inline src/js/plugin.js]: This is an `else if` because if a plugin exists, it will have a method on the `Player.prototype`, but if the plugin _does not exist_ **and** the name matches a `Player.prototype` method, it's an actual player method.

Comment (gkatsev) [inline src/js/plugin.js]: This sounds like a reasonable compromise.

**Comment by gkatsev** [comment]
This should be rebased against master, will make tests run better.

File: src/js/plugin.js [hunk 2]
```diff
@@ -305,8 +306,10 @@ class Plugin {
       throw new Error(`Illegal plugin name, "${name}", must be a string, was ${typeof name}.`);
     }
 
-    if (pluginExists(name) || Player.prototype.hasOwnProperty(name)) {
-      throw new Error(`Illegal plugin name, "${name}", already exists.`);
+    if (pluginExists(name)) {
+      log.warn(`A plugin named "${name}" already exists. You may want to avoid re-registering plugins!`);
```
Comment (gkatsev) [inline src/js/plugin.js]: @misteroneill when you get the chance, can you make another PR that silences this warning in the plugin tests?
